opengrok-tools/src/main/python/opengrok_tools/utils/mirror.py

```diff
@@ -200,6 +200,83 @@ def get_project_properties(project_config, project_name, hookdir):
         use_proxy, ignored_repos
 
 
+def process_hook(hook_ident, hook, source_root, project_name, proxy,
+                 hook_timeout):
+    """
+    :param hook_ident: ident of the hook to be used in log entries
+    :param hook: hook
+    :param source_root: source root path
+    :param project_name: project name
+    :param proxy: proxy or None
+    :param hook_timeout: hook run timeout
+    :return: False if hook failed, else True
+    """
+    if hook:
+        logger = logging.getLogger(__name__)
+
+        logger.info("Running {} hook".format(hook_ident))
+        if run_hook(logger, hook,
+                    os.path.join(source_root, project_name), proxy,
+                    hook_timeout) != SUCCESS_EXITVAL:
+            logger.error("{} hook failed for project {}".
+                         format(hook_ident, project_name))
+            return False
+
+    return True
+
+
+def process_changes(repos, project_name, uri):
+    """
+    :param repos: repository list
+    :param project_name: project name
+    :return: exit code or None
+    """
+    logger = logging.getLogger(__name__)
+
+    changes_detected = False
+
+    # check if the project is a new project - full index is necessary
+    try:
+        r = get(logger, get_uri(uri, 'api', 'v1', 'projects',
+                                urllib.parse.quote_plus(project_name),
+                                'property', 'indexed'))
+        r.raise_for_status()
+        if not bool(r.json()):
+            changes_detected = True
+            logger.debug('Project {} has not been indexed yet'
+                         .format(project_name))
+    except ValueError as e:
+        logger.error('Unable to parse project \'{}\' indexed flag: {}'
+                     .format(project_name, e))
+        return FAILURE_EXITVAL
+    except HTTPError as e:
+        logger.error('Unable to determine project \'{}\' indexed flag: {}'
+                     .format(project_name, e))
+        return FAILURE_EXITVAL
+
+    # check if the project has any new changes in the SCM
+    if not changes_detected:
+        for repo in repos:
+            try:
+                if repo.incoming():
+                    logger.debug('Repository {} has incoming changes'.
+                                 format(repo))
+                    changes_detected = True
+                    break
+            except RepositoryException:
+                logger.error('Cannot determine incoming changes for '
+                             'repository {}'.format(repo))
+                return FAILURE_EXITVAL
+
+    if not changes_detected:
+        logger.info('No incoming changes for repositories in '
+                    'project {}'.
+                    format(project_name))
+        return CONTINUE_EXITVAL
+
+    return None
```

Comment:
I don't like mixing ints and None in return values, maybe SUCCESS_EXITVAL would be better?